lsp-haskell.el

```diff
@@ -263,6 +308,26 @@ and `lsp-haskell-server-args' and `lsp-haskell-server-wrapper-function'."
     ;; This is somewhat irrelevant, but it is listed in lsp-language-id-configuration, so
     ;; we should set something consistent here.
     :language-id "haskell"
+    :download-server-fn (lambda (_client callback error-callback _update?)
+			  (lsp-package-ensure
+			   'haskell-language-server-wrapper
+			   (lambda ()
+			     (let*
+				 ((proj-ghc-ver (lsp-haskell--project-ghc-version))
```

Comment:
So this will only DTRT if it's run from the project directory? I'm not sure what the right behavior for `install-server` should be, especially in a case like this where we effectively have multiple servers. Maybe needing to run it multiple times isn't crazy? Not sure.

(Again, this would be less of an issue if we just downloaded everything up front)